### PR TITLE
import pil2-stark into monorepo

### PR DESCRIPTION
**Pull request by sam0x17**
* imports `pil2-stark`

### CODE REVIEW

File: pil2-stark/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,285 @@
+{
```
Comment (eduadiez) [inline pil2-stark/.vscode/launch.json:1]: I think we should delete or clean this file, most of the things here only apply to the old prover but not necessary for the library library, I've also seen some "rick" stuff here 😅

File: pil2-stark/.vscode/tasks.json
Comment (eduadiez) [inline pil2-stark/.vscode/tasks.json:1]: I think we should delete or rewrite this file too

**Review by eduadiez — APPROVED** [review]

File: pil2-stark/Makefile [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+SRC_DIRS := ./src
+SETUP_DIRS := ./src/rapidsnark
+SETUP_DPNDS_DIR := src/ffiasm
+WITNESS_LIB := ./witness_lib
+
+
+CXX := g++
+AS := nasm
+CXXFLAGS := -std=c++17 -Wall -pthread -flarge-source-files -Wno-unused-label -rdynamic #-Wfatal-errors
+LDFLAGS := -lprotobuf -lsodium -lgpr -lpthread -lpqxx -lpq -lgmp -lstdc++ -lgmpxx -lsecp256k1 -lcrypto -luuid -fopenmp -liomp5
```
Comment (eduadiez) [inline pil2-stark/Makefile:16]: @rickb80 should we remove some dependencies here?

Comment (rickb80) [reply pil2-stark/Makefile:16]: yes, I'll clean this.

**Review by eduadiez — APPROVED** [review]

**Review by RogerTaule — APPROVED** [review]
LGTM

**Pull request merged**